### PR TITLE
Gh 659 consistent edge dir

### PR DESCRIPTION
**Pull request by p013570**
The main change was to reorder the source and destination of undirected edges to ensure the source is 'less than' the destination.

I have removed the magic operation option - "return_matched_id_as_edge_source" and instead added a matchedVertex field to Edge and EdgeSeed. This will be set when a GetElements query is done, so users will now be able to see which end of the edge was matched by a seed.

### CODE REVIEW

File: core/operation/src/main/java/uk/gov/gchq/gaffer/operation/data/EdgeSeed.java [hunk 1]
```diff
@@ -56,58 +77,76 @@ public Object getSource() {
...
+        return matchedVertex;
     }
 
-    @Override
-    public void setDirectedType(final DirectedType directed) {
-        if (null == directed) {
-            this.directed = DirectedType.EITHER;
-        } else {
-            this.directed = directed;
+    private void orderVertices() {
```
Comment (t616178) [inline core/operation/src/main/java/uk/gov/gchq/gaffer/operation/data/EdgeSeed.java]: Could this method be moved to EdgeId to reduce duplication? It's shared between the EdgeSeed and Edge classes.

File: core/operation/src/main/java/uk/gov/gchq/gaffer/operation/data/EdgeSeed.java [hunk 2]
```diff
@@ -56,58 +77,76 @@ public Object getSource() {
...
+        if (!DirectedType.isDirected(directed)) {
+            if (VERTEX_COMPARATOR.compare(source, destination) > 0) {
+                swapVertices();
+            }
         }
     }
 
+    private void swapVertices() {
```
Comment (t616178) [inline core/operation/src/main/java/uk/gov/gchq/gaffer/operation/data/EdgeSeed.java]: Could this method be moved to EdgeId to reduce duplication? It's shared between the EdgeSeed and Edge classes.

File: store-implementation/accumulo-store/src/test/java/uk/gov/gchq/gaffer/accumulostore/key/core/AbstractCoreKeyAccumuloElementConverterTest.java [hunk 3]
```diff
@@ -60,53 +58,76 @@ public void shouldReturnOverriddenSerialiseNull() throws Exception {
...
+        assertEquals(AbstractCoreKeyAccumuloElementConverter.EdgeDirection.DIRECTED, direction);
     }
 
-    private class TestImplementation extends AbstractCoreKeyAccumuloElementConverter {
-        public TestImplementation(Schema schema) {
-            super(schema);
-        }
+    @Test
+    public void shouldDeserialiseSourceDestinationValuesIncorrectWayRound() {
+        // Given
```
Comment (t616178) [inline store-implementation/accumulo-store/src/test/java/uk/gov/gchq/gaffer/accumulostore/key/core/AbstractCoreKeyAccumuloElementConverterTest.java]: Is this an extra character that has crept in? I've seen a few of these dotted around the test classes.

File: store-implementation/hbase-store/src/main/java/uk/gov/gchq/gaffer/hbasestore/serialisation/ElementSerialisation.java [hunk 4]
```diff
@@ -42,13 +43,20 @@
...
 import java.util.Arrays;
 import java.util.Iterator;
-import java.util.Map;
 
 public class ElementSerialisation {
     private static final Logger LOGGER = LoggerFactory.getLogger(ElementSerialisation.class);
 
     private final Schema schema;
 
+    protected enum EdgeDirection {
```
Comment (t616178) [inline store-implementation/hbase-store/src/main/java/uk/gov/gchq/gaffer/hbasestore/serialisation/ElementSerialisation.java]: It feels like we have declarations for variations on this "EdgeDirection" enum in a few different places - would it be better to have one implementation in the core module?

File: core/operation/src/main/java/uk/gov/gchq/gaffer/operation/data/EdgeSeed.java (hunk 2 above)
Comment (p013570) [inline core/operation/src/main/java/uk/gov/gchq/gaffer/operation/data/EdgeSeed.java]: Same as above - the method needs to be private and the swapVertices needs access to set the values of source/destination/matchedVertex.

File: core/operation/src/main/java/uk/gov/gchq/gaffer/operation/data/EdgeSeed.java (hunk 1 above)
Comment (p013570) [inline core/operation/src/main/java/uk/gov/gchq/gaffer/operation/data/EdgeSeed.java]: The method needs to be private and the swapVertices needs access to set the values of source/destination/matchedVertex.

File: store-implementation/hbase-store/src/main/java/uk/gov/gchq/gaffer/hbasestore/serialisation/ElementSerialisation.java (hunk 4 above)
Comment (p013570) [inline store-implementation/hbase-store/src/main/java/uk/gov/gchq/gaffer/hbasestore/serialisation/ElementSerialisation.java]: OK good idea, I've moved it into the data module.

**Comment by p013570** [comment]
@t616178 - thanks. I've made your suggested changes - please can you approve it if you're happy with  it.

**Pull request merged**